https://www.pythonfordatascience.org/anova-python/#anova-test

# ONE-WAY ANOVA

### ONE-WAY ANOVA WITH PYTHON

Don't forget to check the assumptions before interpreting the results!


In [2]:
import pandas as pd
import researchpy as rp

In [3]:
df = pd.read_csv("https://raw.githubusercontent.com/researchpy/Data-sets/master/difficile.csv")
df.drop('person', axis= 1, inplace= True)

# Recoding value from numeric to string
df['dose'].replace({1: 'placebo', 2: 'low', 3: 'high'}, inplace= True)

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 2 columns):
dose      15 non-null object
libido    15 non-null int64
dtypes: int64(1), object(1)
memory usage: 244.0+ bytes


In [4]:
rp.summary_cont(df['libido'])

Variable     N    Mean      SD      SE  95% Conf.  Interval
0   libido  15.0  3.4667  1.7674  0.4563     2.4879    4.4454

In [5]:
rp.summary_cont(df['libido'].groupby(df['dose']))

N  Mean      SD      SE  95% Conf.  Interval
dose                                                 
high     5   5.0  1.5811  0.7071     3.0368    6.9632
low      5   3.2  1.3038  0.5831     1.5811    4.8189
placebo  5   2.2  1.3038  0.5831     0.5811    3.8189

#### ONE-WAY ANOVA USING SCIPY.STATS


In [6]:
import scipy.stats as stats

stats.f_oneway(df['libido'][df['dose'] == 'high'],
               df['libido'][df['dose'] == 'low'],
               df['libido'][df['dose'] == 'placebo'])

F_onewayResult(statistic=5.11864406779661, pvalue=0.024694289538222603)

Before the results should be interpreted, the assumptions of the test should be checked. For example purposes, the results will be interpreted before checking the assumptions.

##### Interpretation

A new medication was developed to increase the libido of those who take the medication. The purpose of this study was to test for a difference between the dosage levels. The overall average libido was 3.5 95% CI(2.5, 4.4) with group averages of 2.2 95% CI(0.9, 3.5) for the placebo group; 3.2 95% CI(1.9, 4.5) for the low dose group; and 5.0 95% CI(3.5, 6.5) for the high dose group. There is a statistically significant difference between the groups and their effects the libido, F= 5.12, p-value= 0.0247.

#### ONE-WAY ANOVA USING STATSMODELS